src/transformStateWithClones.js

```diff
@@ -7,7 +7,47 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(stateCopy, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clear(stateCopy);
+        break;
+
+      default:
+        return 'Error';
```

Comment:
[CODE STYLE]: The default case in a switch statement should not return an error directly. Instead, you should throw an Error to ensure that the error is caught and handled appropriately. For example: `throw new Error('Invalid action type');`